### ISSUE TITLE
Inline imports marked as invalid, though supported

### ISSUE THREAD

**Issue by tedbarth** [gajus/eslint-plugin-jsdoc#706]
Version: 32.3.0

The following code will be marked as invalid due to broken support for inline imports for types.
We get `ESLint: Syntax error in type: import("@amcharts/amcharts4/charts").SeriesDataItem(jsdoc/valid-types)` for both inline imports, while they work fine for code inference in VSCode and Intellij Webstorm/Ultimate.

test.js
```javascript
/**
 * Returns a series.
 *
 * @param {import("@amcharts/amcharts4/charts").SeriesDataItem} dataSeries - A series
 * @return {int} The item height
 */
function foo (dataSeries) {
  return dataSeries.itemHeight;
}

/**
 * @param {import('./foo').Foo} foo
 */
function qux(foo) {
}
```

foo.js
```javascript
/**
 * @typedef Foo
 * @property {string} bar
 */
```

.eslintrc.js
```javascript
module.exports = {
  plugins: [
    'jsdoc',
  ],

  rules: {
    'jsdoc/valid-types': 'error',
  },
};
```

package.json
```json
{
  "dependencies": {
    "@amcharts/amcharts4": "4.10.17"
  },
  "devDependencies": {
    "eslint": "7.22.0",
    "eslint-plugin-jsdoc": "32.3.0"
  }
}
```

**Comment by brettz9** [comment]
If you are not using `@typescript-eslint/parser` as your ESLint `parser`, you can add `{jsdoc: {mode: 'typescript'}` to your `settings` in your ESLint config. The syntax you are using is not recognized by the original jsdoc sofware/spec, but it is well-used with TypeScript and tooling, so it is probably a good choice to go for "typescript" mode.

Closing as that should resolve, but feel free to comment further as needed.
